packages/expect-utils/src/utils.ts

```diff
@@ -336,7 +336,8 @@ const isObjectWithKeys = (a: any) =>
   isObject(a) &&
   !(a instanceof Error) &&
   !Array.isArray(a) &&
-  !(a instanceof Date);
+  !(a instanceof Date) &&
+  !(a instanceof Set);
 
 export const subsetEquality = (
```

Comment:
In case of `subset` as a `Set` the `getObjectKeys(subset)` at line 358 returns empty array (`[]`), so the `.every(...)` in the same line is vacuous truth, resulting the comparison not to raise an exception